### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by rlouf**
The tests for the distribution rely on a specific random number generator implementation and seed. These tests broke as the result of a change upstream in aesara 2.6.5 (I suspect the change happened here https://github.com/aesara-devs/aesara/pull/939). This should not happen and we thus need better tests for the distributions.

### CODE REVIEW

**Comment by zoj613** [comment]
If there are no random stream preservation guarantees by the aesara package, then it means we should not rely on the seed for any tests. We should think about how we can verify the output of those distributions without relying on the seed.

**Comment by zoj613** [comment]
Which properties of the multivariate normal do you think we could test for in the unittest @rlouf ?

File: tests/test_dists.py [hunk 1]
```diff
@@ -60,6 +66,9 @@ def test_multivariate_normal_bhattacharya2016():
     np.testing.assert_allclose(got.eval(), expected)
 
 
+@pytest.mark.skip(
+    reason="The test depends on the choice of seed and the implmentation of the random number generator"
```
Comment (zoj613) [inline tests/test_dists.py:70]:
```suggestion
    reason="The test depends on the choice of seed and the implementation of the random number generator"
```

**Comment by rlouf** [comment]
I've added a test to check that the samples' shape is correct for each sampler. Should we merge this so we can rebase the other PRs and open a separate issue?

**Review by brandonwillard — CHANGES_REQUESTED** [review]
We definitely need some simple moment-based tests (e.g. mean and (co)variance).  It's fine if there's _some_ dependence on an RNG state (e.g. a fixed error window/range of acceptance/etc. that may not pass for all RNG states), since that's basically unavoidable anyway, we just don't want to depend on the exact values of a specific RNG state (e.g. tests that only pass for a single/few RNG states).

**Comment by rlouf** [comment]
@zoj613 Could you have a look at the PR? I added some simple moment-based tests but I am not sure about the formulas I used to compute the mean and variance from the parameters in the last two tests.

**Comment by rlouf** [comment]
I simplified the tests that were not passing; we can merge this so we can move forward on the other PRs and open an issue to add more tests on the distributions.

**Review by zoj613 — APPROVED** [review]

**Pull request merged**